trin-storage/src/versioned/utils.rs

```diff
@@ -0,0 +1,235 @@
+use ethportal_api::types::portal_wire::ProtocolId;
+use r2d2::PooledConnection;
+use r2d2_sqlite::SqliteConnectionManager;
+use rusqlite::{named_params, OptionalExtension};
+use trin_metrics::{portalnet::PORTALNET_METRICS, storage::StorageMetricsReporter};
+
+use crate::{error::ContentStoreError, PortalStorageConfig};
+
+use super::{sql, store::VersionedContentStore, ContentType, StoreVersion};
+
+/// Ensures that the correct version of the content store is used (by migrating the content if
+/// that's not the case).
+pub fn create_store<S: VersionedContentStore>(
+    content_type: ContentType,
+    network: ProtocolId,
+    config: PortalStorageConfig,
+) -> Result<S, ContentStoreError> {
+    let conn = config.sql_connection_pool.get()?;
+
+    let old_version = lookup_store_version(&content_type, &conn)?;
+
+    match old_version {
+        Some(old_version) => {
+            // Migrate if version doesn't match
+            if S::version() != old_version {
```

Comment:
Do we want to migrate only from an older version to the new version, i.e. 
```suggestion
            if S::version() > old_version {
```

Reply:
The versions are not strictly ordered. In this context, the `old_version` should be interpreted as "previous version".

If someone tries to migrate from newer to the older version, the `migrate_from` function should fail.